overlord/servicestate/servicestate.go

```diff
@@ -39,7 +39,7 @@ type Instruction struct {
 
 type ServiceActionConflictError struct{ error }
 
-func Control(st *state.State, appInfos []*snap.AppInfo, inst *Instruction) (*state.TaskSet, error) {
+func Control(st *state.State, appInfos []*snap.AppInfo, inst *Instruction, conflictCheck func(*state.Task) bool) (*state.TaskSet, error) {
```

Comment:
I would make this take a context and not the predicate function

Reply:
Done